### PR TITLE
[VCR/Stripe] Tests payment intents for offline payments

### LINKED ISSUE #11890: Stripe test automation
An epic to gather the issues we've accomplished, and the ones which are still pending so testers don't have to manually test Stripe each week / gem bump.

- Done:
- [x] [Integration/API tests] Add VCR to the stack https://github.com/openfoodfoundation/openfoodnetwork/pull/9624
- [x] [Integration/API tests] Improve payment intent tests with VCR calls https://github.com/openfoodfoundation/openfoodnetwork/pull/11742
- [x] [Integration/API tests] Extend payment intent tests i) test error responses from payment attempts made from a list of invalid cards: https://github.com/openfoodfoundation/openfoodnetwork/pull/11810
- [x] #11892
- [x] #11925
- [x] #11949
- [x] #11962
- [x] #12001
- [x] [Integration/API tests] Cards: saving cards; deleting cards ( -> covered within the folder [*/lib/stripe](https://github.com/openfoodfoundation/openfoodnetwork/issues/12001); hence marked as completed.)
- [x] #12164
- [x] #12201
- [x] #12034

- Related:

- [x] [Active Merchant] Refactor/decorator removal https://github.com/openfoodfoundation/openfoodnetwork/issues/7326
- [x] [Active Merchant] gem bump https://github.com/openfoodfoundation/openfoodnetwork/pull/11880

- To do:

~~- [ ] [System tests] Set up browser proxy + VCR for system tests: https://github.com/openfoodfoundation/openfoodnetwork/issues/8957~~
~~- [ ] [System tests] Saving cards on checkout https://github.com/openfoodfoundation/openfoodnetwork/issues/3952~~
~~- [ ] [System tests] Subscriptions (requires setting up a customer and a card)~~

- In code review:
- [x] #12409

### LINKED ISSUE #12409: [Integration/API tests] Customer: Test payment intents as a guest customer; as session customer

### PR DESCRIPTION
**Pull request by filipefurtad0**
#### What? Why?

- Contributes/closes #12409 and #11890.

Subscriptions use payment intents with a set up customer, for offline payments, i.e., payments in which the customer is not online, at the moment of checkout.

Before we were only testing regular payment intents, with a guest customer.

This PR adds tests around payment intents which have a customer previously set up, for offline payments.

Please note, that **this PR does not test OFN code**. It serves as a proof of concept, that we can trigger the correct card responses in both scenarios (with and without a setup customer).

I have not found a Stripe spec directly addressing subscriptions or setting offline payments; I think it would need to be created from scratch.

#### What should we test?

- green build.

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Review by dacook — CHANGES_REQUESTED** [review]
The code looks good, just added one note but perhaps it's more of an opinion.  `±` The linter step failed though.

So this specifies some of the Stripe behaviour, (which I guess is also documented by Stripe). 
Do you intend to keep these, or is the next step to use them as a starting point for creating the OFN specs?

**Comment by filipefurtad0** [comment]
> The linter step failed though.

This is occurring when we introduce too many new lines, also when we re-record Stripe cassettes - [it started on version 10.14.0 I believe](https://github.com/openfoodfoundation/openfoodnetwork/pulls?q=is%3Apr+is%3Aclosed+bump+stripe+).

> Do you intend to keep these, or is the next step to use them as a starting point for creating the OFN specs?

I would say these are good to keep for now, while we don't have specific specs for subscriptions. These tests could potentially catch breaking changes around payments setup for future use, when bumping Stripe versions. But, yes, we can consider removing these, once we write more specific tests.

File: spec/lib/stripe/payment_intent_validator_spec.rb [hunk 1]
```diff
@@ -18,131 +18,290 @@
...
+          let!(:user) { create(:user, email: "apple.customer@example.com") }
+          let(:credit_card) { create(:credit_card, gateway_payment_profile_id: pm_card.id, user:) }
+
+          context "from #{card_type}" do
+            let(:customer) do
+              Stripe::Customer.create({
+                                        name: 'Apple Customer',
+                                        email: 'applecustomer@example.com',
+                                      })
+            end
```
[the remark this answers is not in the document]
Comment (filipefurtad0) [reply spec/lib/stripe/payment_intent_validator_spec.rb]: Thanks @dacook , I've moved the customer block and the let variables outside the shared_examples block.

**Comment by dacook** [comment]
> This is occurring when we introduce too many new lines

That's funny, the diff is indeed large, but most of it is YAML, which Rubocop is configured to ignore.
This will probably solved by [Test all files with Rubocop, not just added code in the diff](https://github.com/openfoodfoundation/openfoodnetwork/pull/12393#top).

**Review by mkllnk — APPROVED** [review]

**Pull request merged**